src/pages/top-stories/index.tsx

```diff
@@ -0,0 +1,34 @@
+import {
+  dehydrate,
+  DehydratedState,
+  HydrationBoundary,
+  QueryClient,
+} from '@tanstack/react-query';
+import { prefetchTopStories } from '@/api/top-stories/queries';
+import { TopStories } from '@/components/top-stories/TopStories';
+
+type Props = {
+  dehydratedState: DehydratedState;
+};
+
+function TopStoriesPage({ dehydratedState }: Props) {
+  return (
+    <HydrationBoundary state={dehydratedState}>
```

Comment:
do you want to put `HydrationBoundary` on each page? why not to move to upper level? (_app or layout)